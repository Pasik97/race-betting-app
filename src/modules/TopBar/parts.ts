import styled from 'styled-components';
import { Link } from 'react-router-dom';
import media from 'utils/media';

export const TopBarWrapper = styled.header`
   display: flex;
   align-items: baseline;
   flex-direction: column;
   color: #1a1a1a;
   box-shadow: rgb(0 0 0 / 64%) 0px 2px 8px -5px;
   padding: 16px 24px;
   margin-bottom: 32px;

   ${media.fromTablet} {
      display: flex;
      flex-direction: row;
      padding: 16px 10%;
   }
`;

export const AppTitle = styled(Link)`
   font-size: 28px;
   line-height: 32px;
   color: #01263A;
   margin: 0;
   font-weight: 700;
   cursor: pointer;
   text-decoration: none;
`;

export const LinksWrapper = styled.div`
   margin-top: 8px;

   ${media.fromTablet} {
      margin-left: 64px;
   }
`;

export const StyledLink = styled(Link)`
   color: #1a1a1a;
   font-size: 21px;
   line-height: 24px;
   text-decoration: none;

   &:hover {
      color: #01263A;
   }
`;

