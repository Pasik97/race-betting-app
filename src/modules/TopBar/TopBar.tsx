import React from 'react';
import * as P from './parts';

const TopBar: React.FC = () => (
   <P.TopBarWrapper>
      <P.AppTitle title="Race betting system" to='/'>Race betting system</P.AppTitle>
      <P.LinksWrapper>
         <P.StyledLink to='/races'>Races</P.StyledLink>
      </P.LinksWrapper>
   </P.TopBarWrapper>
)

export default TopBar;
