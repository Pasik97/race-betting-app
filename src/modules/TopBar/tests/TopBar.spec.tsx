import { BrowserRouter } from "react-router-dom";
import { create } from "react-test-renderer";
import TopBar from "../TopBar";

describe('<TopBar />', () => {
   it('should render header with title as link to home page and link to races page', () => {
      const component = create(<BrowserRouter><TopBar /></BrowserRouter>);

      expect(component.toJSON()).toMatchSnapshot();
   });
});
