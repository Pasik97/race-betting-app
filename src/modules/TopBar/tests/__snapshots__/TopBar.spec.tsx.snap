// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBar /> should render header with title as link to home page and link to races page 1`] = `
<header
  className="sc-bdvvtL dktecm"
>
  <a
    className="sc-gsDKAQ gtcDQK"
    href="/"
    onClick={[Function]}
    title="Race betting system"
  >
    Race betting system
  </a>
  <div
    className="sc-dkPtRN jUPLyL"
  >
    <a
      className="sc-hKwDye caybUZ"
      href="/races"
      onClick={[Function]}
    >
      Races
    </a>
  </div>
</header>
`;
